Source/Moya.swift

```diff
@@ -21,7 +21,33 @@ public protocol TargetType {
     var path: String { get }
     var method: Moya.Method { get }
     var parameters: [String: AnyObject]? { get }
+    var parameterEncoding: ParameterEncoding { get }
     var sampleData: NSData { get }
+
+    func toEndpoint() -> Endpoint
+}
+
+public extension TargetType {
+    func toEndpoint() -> Endpoint {
+        let url = self.baseURL.URLByAppendingPathComponent(self.path).absoluteString
+        return Endpoint(URL: url, sampleResponseClosure: {.NetworkResponse(200, self.sampleData)}, method: self.method, parameters: self.parameters, parameterEncoding: parameterEncoding)
+    }
```

Comment:
But in a protocol extension, if we only have a reference to the type as `TargetType`, won't this _always_ execute this instead of the function specified in the actual type?